lib/jwt/decode.rb

```diff
@@ -9,7 +9,7 @@ module JWT
   # Decoding logic for JWT
   class Decode
     def initialize(jwt, key, verify, options, &keyfinder)
-      raise(JWT::DecodeError, 'Nil JSON web token') unless jwt
+      raise(JWT::DecodeError, "#{jwt.class} JSON web token") unless jwt.class == String
```

Comment:
Would suggest using the `Object#is_a?`to check the type of the given parameter. The method also takes inheritance into consideration.

Also think it would solve the sourceleve-bot whining.

Reply:
Im wondering if this could be just a `raise TypeError, 'JSON web token is expected to be a String #{jwt.class} given' unless jwt.is_a?(String)`


Maybe to not brake backwards compatibility:
`raise JWT::DecodeError, 'JSON web token is expected to be a String #{jwt.class} given' unless jwt.is_a?(String)`